src/Utils.ts

```diff
@@ -0,0 +1,103 @@
+
+export class Utils {
+    /**
+     * Returns all the tags for requested state
+     *
+     * @returns {String} all the tags for requested state
+     */
+    public static getTagsForState(state : string) : string {
+        let tags = '';
+        if (state === Constants.AUTHORING) {
+            const clientLibs = this.generateClientLibsUrl();
+            clientLibs.forEach(clientlib => {
+                if (clientlib.endsWith('.js')) {
+                    tags = tags.concat(this.generateElementString('script', 'src', clientlib));
+                }
+                else if (clientlib.endsWith('.css')) {
+                    tags = tags.concat(this.generateElementString('stylesheet', 'href', clientlib));
+                }
+            });
+        }
+        return tags;
+    }
+
+    /**
+     * Returns string value of all the concatenated tags
+     *
+     * @returns {String} concatenated tags
+     */
+    private static generateElementString(tagType : string, attr : string, attrValue : string) : string {
+        let tag = '';
+        if (tagType === 'script') {
+            tag = `<script ${attr}='${attrValue}'></script>`;
+        }
+        else if (tagType === 'stylesheet') {
+            tag = `<link ${attr}='${attrValue}'/>`;
+        }
+        return tag;
+    }
+
+    /**
+     * Returns if requested state is active
+     *
+     * @returns {boolean} the result of the check of the state
+     */
+    public static isState(state : string) : boolean {
+        if (state === Constants.AUTHORING) {
+            const viaMetaProperty = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === Constants.AEM_MODE_EDIT;
+            const viaQueryParam = PathUtils.isBrowser() && (Utils.getEditParam() === Constants.AEM_MODE_EDIT);
+
+            return viaMetaProperty || viaQueryParam;
+        }
+        return false;
+    }
+
+    /**
+     * Returns query parameter value for aem mode.
+     *
+     * @returns {String | null} the result of the check of the Editor mode
+     */
+    public static getEditParam() : string | null {
+        return new URL(PathUtils.getCurrentURL()).searchParams.get(Constants.AEM_MODE_KEY);
+    }
+
+    /**
+     * Generates clientlib url
+     * @returns {string[]} all the clientlib urls
+     */
+    public static generateClientLibsUrl() : string[] {
+        const clientlibs: string[] =  Array<string>();
+        const path = '/etc.clientlibs/cq/gui/components/authoring/editors/clientlibs/internal/';
+        const domain = this.getDomain();
+        clientlibs.push(`${domain}${path}page.js`);
+        clientlibs.push(`${domain}${path}page.css`);
+        clientlibs.push(`${domain}${path}pagemodel/messaging.js`);
+        clientlibs.push(`${domain}${path}messaging.js`);
+        return clientlibs;
+    }
+
+
+    /**
+     * Gets the domain name of API host
+     * @returns {string} API host
+     */
+    public static getDomain() : string{
+        return ModelManager._domain;
```

Comment:
Can we test this? There's this side effect initialisation of this _domain element so we probably should at least test it.